### PR TITLE
Add example: custom title bar

### PR DESCRIPTION
**Pull request by ski0090**
Add an example of creating a custom title bar.

### CODE REVIEW

File: examples/header_bar/title_bar.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        let menu = StackSwitcher::new();
+        let header = HeaderBar::builder()
+            .custom_title(&menu)
+            .show_close_button(true)
+            .build();
+        header.add(&Label::new(Some("Custom title")));
+        Self { header, menu }
+    }
+
+    pub fn set_stack(&self, stack: gtk::Stack) {
```
Comment (bilelmoussaoui) [inline examples/header_bar/title_bar.rs:20]: Why take this by value if you only need a ref?

File: examples/header_bar/main.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    let title_bar = TitleBar::new();
+    window.set_titlebar(Some(title_bar.header()));
+    title_bar.set_stack(stack);
+
+    window.show_all();
+}
+
+fn main() {
+    gio::resources_register_include!("compiled.gresource").unwrap();
```
Comment (bilelmoussaoui) [inline examples/header_bar/main.rs:36]:
```suggestion
```
doesn't seem to be used for anything?

File: examples/header_bar/main.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+use gtk::Label;
+use gtk::Stack;
+use gtk::{ApplicationWindow, WindowPosition};
+use title_bar::TitleBar;
+
+fn build_ui(application: &gtk::Application) {
+    let window = ApplicationWindow::new(application);
+
+    window.set_title("HeaderBar example");
+    window.set_position(WindowPosition::Center);
```
Comment (bilelmoussaoui) [inline examples/header_bar/main.rs:15]: could have used a builder pattern here similar to others

File: examples/header_bar/main.rs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+mod title_bar;
+
+use gtk::gio;
+use gtk::prelude::*;
+use gtk::CheckButton;
+use gtk::Label;
+use gtk::Stack;
+use gtk::{ApplicationWindow, WindowPosition};
```
Comment (bilelmoussaoui) [inline examples/header_bar/main.rs:8]:
```suggestion
use gtk::{ApplicationWindow, CheckButton, Label, Stack, WindowPosition};
```

Comment (ski0090) [reply examples/header_bar/main.rs:8]: Is it not necessary to include prelude?

```rust
use gtk::prelude::*;
use gtk::{ApplicationWindow, CheckButton, Label, Stack, WindowPosition};
```
vs

```rust
use ;
use gtk::{gtk::prelude::*, ApplicationWindow, CheckButton, Label, Stack, WindowPosition};
```

Comment (bilelmoussaoui) [reply examples/header_bar/main.rs:8]: my code suggestion doesn't remove the prelude, it is required yes.

File: examples/header_bar/title_bar.rs (hunk 1 above)
Comment (ski0090) [reply examples/header_bar/title_bar.rs:20]: I don't think I considered the life time of ref. I modified it.

File: examples/header_bar/main.rs (hunk 2 above)
Comment (ski0090) [reply examples/header_bar/main.rs:36]: That's right

File: examples/header_bar/main.rs (hunk 3 above)
Comment (ski0090) [reply examples/header_bar/main.rs:15]: Changed to Builder pattern.

**Comment by ski0090** [comment]
> Add an example of creating a custom title bar.

@GuillaumeGomez You were registered as a reviewer, but you disappeared when I pressed Refresh Review. Can I review it?

File: examples/header_bar/main.rs (hunk 4 above)
Comment (ski0090) [reply examples/header_bar/main.rs]: Modified in the title bar and main!

**Comment by ski0090** [comment]
@bilelmoussaoui 
Can i merge?

File: examples/header_bar/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# HeaderBar example
+
+This example demonstrates how to use HeaderBar in Windows.
```
Comment (GuillaumeGomez) [inline examples/header_bar/README.md:3]:
```suggestion
This example demonstrates how to use HeaderBar in windows.
```

To not be confused with the OS.

Comment (GuillaumeGomez) [reply examples/header_bar/README.md]: Yep exactly.

**Comment by GuillaumeGomez** [comment]
Thanks and sorry for the wait!

**Pull request merged**